✨ Source Typeform: Add OAuth support (#27240)
airbytehq/airbyte: docs/integrations/sources/typeform.md

## Prerequisites

- [Typeform Account](https://www.typeform.com/)

> - Start Date - Date to start fetching Responses stream data from.

Reviewer: They don't need this here, the pre-requisites section is for things they need before they even start setting up the connector
```suggestion
```

- Form IDs (Optional) - If you want to sync data for specific forms, you'll need to have the IDs of those forms. If you want to sync data for all forms in your account you don't need any IDs. Form IDs can be found in the URLs to the forms in Typeform Admin Panel (for example, for URL `https://admin.typeform.com/form/12345/` a `12345` part would your Form ID)
<!-- env:cloud -->
**For Airbyte Cloud:**

- OAuth
<!-- /env:cloud -->

<!-- env:oss -->
**For Airbyte Open Source:**

- Personal Access Token (see [personal access token](https://www.typeform.com/developers/get-started/personal-access-token/))
<!-- /env:oss -->

## Setup guide

> ### Step 1: Set up Typeform

Reviewer: ```suggestion
### Step 1: Obtain an API token 
```

<!-- env:oss -->
**For Airbyte Open Source:**
To get the API token for your application follow this [steps](https://developer.typeform.com/get-started/personal-access-token/)
* Log in to your account at Typeform.
* In the upper-right corner, in the drop-down menu next to your profile photo, click My Account.
* In the left menu, click Personal tokens.
* Click Generate a new token.
* In the Token name field, type a name for the token to help you identify it.
* Choose needed scopes \(API actions this token can perform - or permissions it has\). See [here](https://www.typeform.com/developers/get-started/scopes/) for more details on scopes.
* Click Generate token.
<!-- /env:oss -->

<!-- env:cloud -->
**For Airbyte Cloud:**

> * Authorize to your [Typeform Account](https://www.typeform.com/), it will be used during next step to authorize using OAuth.

Reviewer: ```suggestion
This step is not needed in Airbyte Cloud. Skip to the next step. 
```

## Prerequisites

- [Typeform Account](https://www.typeform.com/)
- Start Date - Date to start fetching Responses stream data from.

> - form_ids (Optional) - List of Form Ids to sync. If not passed - sync all account`s forms.

Reviewer: ```suggestion
- Form IDs (Optional) - If you want to sync data for specific forms, you'll need to have the IDs of those forms. If you want to sync data for all forms in your account you don't need any IDs. 
```

Reviewer: also, can you tell the user where to get the form IDs?

Author: added your suggestion, and added explanation about Form IDs

<!-- env:cloud -->
**For Airbyte Cloud:**

> - Personal Access Token (see [personal access token](https://www.typeform.com/developers/get-started/personal-access-token/))

Reviewer: I recommend we delete this section for cloud since users should just use oauth

* In the left menu, click Personal tokens.
* Click Generate a new token.
* In the Token name field, type a name for the token to help you identify it.

> * Choose needed scopes \(API actions this token can perform - or permissions it has\). See [here](https://www.typeform.com/developers/get-started/scopes/) for more details on scopes.

Reviewer: this step should only be shown in OSS not cloud

Author: moved this section to only be shown in OSS

**For Airbyte Cloud:**

1. [Log into your Airbyte Cloud](https://cloud.airbyte.com/workspaces) account.
2. In the left navigation bar, click **Sources**. In the top-right corner, click **+ New Source**.
3. On the source setup page, select **Typeform** from the Source type dropdown and enter a name for this connector.
4. Click `Authenticate your Typeform account` by selecting Oauth or Personal Access Token for Authentication.
5. Log in and Authorize to the Typeform account.

> 6. **Start date** - Date to start fetching Responses stream data from.

Reviewer: can we make start date optional? why should it be required?

Author: I have made start_date optional, as Typeform does not have a default date in case we fetch API without it, I have set default date to be a date a year from now.
